lib/qops/environment.rb

```diff
@@ -31,20 +31,83 @@ def self.print_with_colour(message, level = :normal)
       end
     end
 
-    def initialize
-      %w[deploy_type region stack_id app_name].each do |v|
+    def initialize(profile: nil, force_config: false)
+      @_aws_config = { region: configuration.region }
+      @_aws_config[:profile] = profile unless profile.nil?
+      @_force_config = force_config
+      puts Rainbow("using aws profile #{profile}").bg(:black).green unless profile.nil?
+
+      %w[deploy_type region app_name].each do |v|
         fail "Please configure #{v} before continuing." unless option?(v)
       end
 
+      # if being forced to use config , then stack_id is a requirement
+      fail 'Please configure stack_id or stack_name before continuing' unless option?('stack_id') || option?('stack_name')
+
       begin
-        opsworks.config.credentials.credentials
+        opsworks.config.credentials.credentials unless profile
       rescue => e # rubocop:disable Lint/RescueWithoutErrorClass
         raise "There may be a problem with your aws credentials. Please correct with `aws configure`. Error: #{e}"
       end
     end
 
-    def application_id
-      configuration.application_id unless configuration.application_id.blank?
+    def identity_from_config
+      configuration.stack_id ? :stack_id : :stack_name
+    end
+
+    def find_stack(options = {})
+      return @_stack if @_stack
+      key = if !options[:name].nil?
+              :name
+            elsif !options[:stack_id].nil?
+              :stack_id
+            else
+              id = identity_from_config
+              msg = Rainbow("Using opsworks.yml config #{id}: #{configuration.send(id)}")
+              puts(msg.bg(:black).green)
+              id
+            end
+      value = options[key] || configuration.send(key)
+      key = :name if key == :stack_name
+      stack = opsworks.describe_stacks.stacks.find { |s| s.send(key) == value }
+      unless stack
+        puts Rainbow("Could not find stack with #{key} = #{value}").bg(:black).red
+        exit(-1)
+      end
+      @_stack = stack
+    end
```

Comment:
this function is a little big and hard to understand, do you mind to break it into several short and easy to understand functions?